Extranous argument label in .setup method

After upgrading to 0.10.1 I get a compile/build error when trying to Archive my app. It says I have an extranous argument label in my .setup method and suggests I remove `viewControllers:` - if I do, it suggests I add `viewControllers:`. I downgraded to 0.9.1 and now it works again.

I could build and run the app on simulator but not archive it. Very odd?

Method called like: `pagingMenuController.setup(viewControllers: vcs, options: options)`

Where `vcs` is an array of `UIViewController` and options is an instance of `PagingMenuOptions`.

Hey

I just tried to just press "Build" in X-Code, and I get this error: 

Undefined symbols for architecture arm64:
  "PagingMenuController.PagingMenuOptions.__allocating_init () -> PagingMenuController.PagingMenuOptions", referenced from:
      Nyx.NewTableViewController.viewDidLoad () -> () in NewTableViewController.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

Where NewTableViewController is my own class of course, but clearly this has something to do with this library.

The odd thing is I can clean, archive and run the app. I just can't press "Build"? I tried deleting DerivedData folder as well.

--- a/PagingMenuController.podspec
+++ b/PagingMenuController.podspec
@@ -9,7 +9,7 @@
 
 Pod::Spec.new do |s|
   s.name             = "PagingMenuController"
-  s.version          = "0.10.1"
+  s.version          = "0.10.2"
   s.summary          = "A paging view with customizable menu"
   s.homepage         = "https://github.com/kitasuke/PagingMenuController"
   # s.screenshots     = "www.example.com/screenshots_1", "www.example.com/screenshots_2"

--- a/Pod/Classes/MenuItemView.swift
+++ b/Pod/Classes/MenuItemView.swift
@@ -180,17 +180,26 @@ public class MenuItemView: UIView {
     private func layoutImageView() {
         guard let image = menuImageView.image else { return }
         
-        let viewsDictionary = ["imageView": menuImageView]
-        let horizontalConstraints = NSLayoutConstraint.constraintsWithVisualFormat("H:|[imageView]|", options: [], metrics: nil, views: viewsDictionary)
-        let verticalConstraints = NSLayoutConstraint.constraintsWithVisualFormat("V:|[imageView]|", options: [], metrics: nil, views: viewsDictionary)
-        NSLayoutConstraint.activateConstraints(horizontalConstraints + verticalConstraints)
-        
         let width: CGFloat
+        let margin: CGFloat
         switch options.menuDisplayMode {
-        case .SegmentedControl: width = UIApplication.sharedApplication().keyWindow!.bounds.size.width / CGFloat(options.menuItemCount)
-        default: width = image.size.width
+        case .SegmentedControl:
+            width = UIApplication.sharedApplication().keyWindow!.bounds.size.width / CGFloat(options.menuItemCount)
+            margin = (width - image.size.width) / 2
+        default:
+            width = image.size.width + horizontalMargin * 2
+            margin = horizontalMargin
         }
-        widthConstraint = NSLayoutConstraint(item: menuImageView, attribute: .Width, relatedBy: .Equal, toItem: nil, attribute: .Width, multiplier: 1.0, constant: width)
+        
+        let metrics = ["width": image.size.width, "margin": margin]
+        let viewsDictionary = ["imageView": menuImageView]
+        let horizontalConstraints = NSLayoutConstraint.constraintsWithVisualFormat("H:|-margin-[imageView(width)]-margin-|", options: [], metrics: metrics, views: viewsDictionary)
+        
+        NSLayoutConstraint.activateConstraints([
+            NSLayoutConstraint(item: menuImageView, attribute: .CenterY, relatedBy: .Equal, toItem: self, attribute: .CenterY, multiplier: 1.0, constant: 0.0)
+            ] + horizontalConstraints)
+        
+        widthConstraint = NSLayoutConstraint(item: self, attribute: .Width, relatedBy: .Equal, toItem: nil, attribute: .Width, multiplier: 1.0, constant: width)
         widthConstraint.active = true
     }
     

--- a/Pod/Classes/MenuItemView.swift
+++ b/Pod/Classes/MenuItemView.swift
@@ -181,26 +181,22 @@ public class MenuItemView: UIView {
         guard let image = menuImageView.image else { return }
         
         let width: CGFloat
-        let margin: CGFloat
         switch options.menuDisplayMode {
         case .SegmentedControl:
             width = UIApplication.sharedApplication().keyWindow!.bounds.size.width / CGFloat(options.menuItemCount)
-            margin = (width - image.size.width) / 2
         default:
             width = image.size.width + horizontalMargin * 2
-            margin = horizontalMargin
         }
         
-        let metrics = ["width": image.size.width, "margin": margin]
-        let viewsDictionary = ["imageView": menuImageView]
-        let horizontalConstraints = NSLayoutConstraint.constraintsWithVisualFormat("H:|-margin-[imageView(width)]-margin-|", options: [], metrics: metrics, views: viewsDictionary)
+        widthConstraint = NSLayoutConstraint(item: self, attribute: .Width, relatedBy: .Equal, toItem: nil, attribute: .Width, multiplier: 1.0, constant: width)
         
         NSLayoutConstraint.activateConstraints([
-            NSLayoutConstraint(item: menuImageView, attribute: .CenterY, relatedBy: .Equal, toItem: self, attribute: .CenterY, multiplier: 1.0, constant: 0.0)
-            ] + horizontalConstraints)
-        
-        widthConstraint = NSLayoutConstraint(item: self, attribute: .Width, relatedBy: .Equal, toItem: nil, attribute: .Width, multiplier: 1.0, constant: width)
-        widthConstraint.active = true
+            NSLayoutConstraint(item: menuImageView, attribute: .CenterX, relatedBy: .Equal, toItem: self, attribute: .CenterX, multiplier: 1.0, constant: 0.0),
+            NSLayoutConstraint(item: menuImageView, attribute: .CenterY, relatedBy: .Equal, toItem: self, attribute: .CenterY, multiplier: 1.0, constant: 0.0),
+            NSLayoutConstraint(item: menuImageView, attribute: .Width, relatedBy: .Equal, toItem: nil, attribute: .Width, multiplier: 1.0, constant: image.size.width),
+            NSLayoutConstraint(item: menuImageView, attribute: .Height, relatedBy: .Equal, toItem: nil, attribute: .Height, multiplier: 1.0, constant: image.size.height),
+            widthConstraint
+            ])
     }
     
     private func layoutDivider() {

--- a/Pod/Classes/PagingMenuController.swift
+++ b/Pod/Classes/PagingMenuController.swift
@@ -264,9 +264,9 @@ public class PagingMenuController: UIViewController {
         
         let previousViewController = currentViewController
         
+        delegate?.willMoveToPageMenuController?(currentViewController, previousMenuController: previousViewController)
         updateCurrentPage(page)
         currentViewController = pagingViewControllers[currentPage]
-        delegate?.willMoveToPageMenuController?(currentViewController, previousMenuController: previousViewController)
         menuView?.moveToMenu(page)
         
         let duration = animated ? options.animationDuration : 0

--- a/Pod/Classes/PagingMenuController.swift
+++ b/Pod/Classes/PagingMenuController.swift
@@ -8,9 +8,6 @@
 
 import UIKit
 
-public protocol MenuControllerType {}
-extension UIViewController: MenuControllerType {}
-
 @objc public protocol PagingMenuControllerDelegate: class {
     optional func willMoveToPageMenuController(menuController: UIViewController, previousMenuController: UIViewController)
     optional func didMoveToPageMenuController(menuController: UIViewController, previousMenuController: UIViewController)
@@ -113,23 +110,23 @@ public class PagingMenuController: UIViewController {
 
     // MARK: - Lifecycle
     
-    public init<Element: MenuControllerType>(menuControllerTypes: [Element], options: PagingMenuOptions) {
+    public init(viewControllers: [UIViewController], options: PagingMenuOptions) {
         super.init(nibName: nil, bundle: nil)
         
-        setup(menuControllerTypes, options: options)
+        setup(viewControllers, options: options)
     }
     
-    convenience public init<Element: MenuControllerType>(menuControllerTypes: [Element]) {
-        self.init(menuControllerTypes: menuControllerTypes, options: PagingMenuOptions())
+    convenience public init(viewControllers: [UIViewController]) {
+        self.init(viewControllers: viewControllers, options: PagingMenuOptions())
     }
     
-    public init<Element: MenuItemType>(menuItemTypes: [Element], options: PagingMenuOptions) {
+    public init(menuItemTypes: [MenuItemType], options: PagingMenuOptions) {
         super.init(nibName: nil, bundle: nil)
         
         setup(menuItemTypes, options: options)
     }
     
-    convenience public init<Element: MenuItemType>(menuItemTypes: [Element]) {
+    convenience public init(menuItemTypes: [MenuItemType]) {
         self.init(menuItemTypes: menuItemTypes, options: PagingMenuOptions())
     }
     
@@ -175,9 +172,9 @@ public class PagingMenuController: UIViewController {
     
     // MARK: - Public
     
-    public func setup<Element: MenuControllerType>(menuControllerTypes: [Element], options: PagingMenuOptions) {
+    public func setup(viewControllers: [UIViewController], options: PagingMenuOptions) {
         self.options = options
-        pagingViewControllers = menuControllerTypes.map { $0 as! UIViewController }
+        pagingViewControllers = viewControllers
         visiblePagingViewControllers.reserveCapacity(visiblePagingViewNumber)
         
         // validate
@@ -217,7 +214,7 @@ public class PagingMenuController: UIViewController {
         layoutPagingViewControllers()
     }
     
-    public func setup<Element: MenuItemType>(menuItemTypes: [Element], options: PagingMenuOptions) {
+    public func setup(menuItemTypes: [MenuItemType], options: PagingMenuOptions) {
         self.options = options
         currentPage = options.defaultPage
         options.menuComponentType = .MenuView

